File: src/main/java/com/vyperplugin/psi/VyperFile.java
package com.vyperplugin.psi;

import com.intellij.extapi.psi.PsiFileBase;
import com.intellij.openapi.fileTypes.FileType;
import com.intellij.psi.FileViewProvider;
import com.vyperplugin.Vyper;
import com.vyperplugin.VyperFileType;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;

public class VyperFile extends PsiFileBase {

    public VyperFile(@NotNull FileViewProvider viewProvider) {
        super(viewProvider,  Vyper.INSTANCE);
    }

    @NotNull
    @Override
    public FileType getFileType() {
        return VyperFileType.INSTANCE;
    }

    @Override
    public String toString() {
        return "Vyper File";
    }

    @Override
    public Icon getIcon(int flags) {
        return super.getIcon(flags);
    }
}

File: src/main/kotlin/com/vyperplugin/psi/VyperFile.kt
package com.vyperplugin.psi

import com.intellij.extapi.psi.PsiFileBase
import com.intellij.openapi.fileTypes.FileType
import com.intellij.psi.FileViewProvider
import com.intellij.psi.util.PsiTreeUtil
import com.vyperplugin.VyperFileType
import com.vyperplugin.VyperLanguage

class VyperFile(viewProvider: FileViewProvider) : PsiFileBase(viewProvider, VyperLanguage) {

    override fun getFileType(): FileType {
        return VyperFileType.INSTANCE
    }

    override fun toString(): String {
        return "Vyper File"
    }

    fun getStatements(): List<VyperElement> =
        PsiTreeUtil.getChildrenOfAnyType(this, VyperElement::class.java)
}
